en/concept/events_interface.md

```diff
@@ -0,0 +1,133 @@
+    For the majority of cases you can pass a single log level, and this will be used for both exernal and internal cases.
+	There are cases it makes sense to have two different log levels.
+	For example an RTL failsafe action: the user should see it as Warning/Error, whereas in the log, it is an expected system response, so it can be set to `Info`.
+- **Event Message**:
+  - Single-line, short description of the event.
+    It may contain arguments (see below).
+- **Event Description**:
+  - detailed, optional event description.
+  - can be multiple lines/paragraphs.
+
+#### Arguments and Enums
+
+Events can have a fixed set of arguments that can be shown in the message or description.
+
+Valid types: `uint8_t`, `int8_t`, `uint16_t`, `int16_t`, `uint32_t`, `int32_t`, `uint64_t`, `int64_t` and `float`.
+
+It is also possible to define and use custom enumerations.
+These are defined in [src/lib/events/enums.json](https://github.com/PX4/PX4-Autopilot/blob/master/src/lib/events/enums.json), and can then be used as event argument in the form of `events::send<events::px4::enums::my_enum_t>(...)`.
+
+#### Text format
+
+Text format for event message description:
+- characters can be escaped with \\
+
+  These have to be escaped: '\\\\', '\\<', '\\{'.
+- supported tags:
+  - Profiles: `<profile name="[!]NAME">CONTENT</profile>`
+
+    `CONTENT` will only be shown if the name matches the configured profile.
+	This can be used for example to hide developer information from end-users.
+  - URLs: `<a [href="URL"]>CONTENT</a>`.
+    If href is not set, use `CONTENT` as `URL`
+  - Parameters: `<param>PARAM_NAME</param>`
+  - no nested tags of the same type are allowed
+- arguments: following python syntax, with 1-based indexing (instead of 0)
+  - general form: `{ARG_IDX[:.NUM_DECIMAL_DIGITS][UNIT]}`
+
+    UNIT:
+      - m: horizontal distance in meters
+      - m_v: vertical distance in meters
+      - m^2: area in m^2
+      - m/s: speed in m/s
+      - C: temperature in degrees celcius
+
+## Logging
+Events are logged according to the internal log level, and [Flight Review](../log/flight_review.md) displays events.
+
+:::note
+Flight review downloads metadata based on PX4 master, so if a definition is not yet on master, it will only be able to display the event ID.
+:::
+
+
+## Implementation
+During build, only the event ID, log level(s) and arguments (if any) are added to the binary by the compiler.
+At the same time, a python script scans the whole source code for event calls, extracts the metadata and assembles a JSON metadata file.
+
+As for parameter metadata, the events metadata is [compiled into the firmware](../advanced/parameters_and_configurations.md#publishing-parameter-metadata-to-a-gcs), and made available to ground stations via the [MAVLink Component Information service](https://mavlink.io/en/services/component_information.html).
```

Comment:
In the same way as for parameter metadata

Reply:
Right